### PR TITLE
Look for cross compile folders too

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/conda_forge_ci_setup/upload_or_check_non_existence.py [hunk 1]
```diff
@@ -169,6 +170,14 @@ def upload_or_check(
                 os.path.join(conda_build.config.croot, conda_build.config.subdir))
         ]
     )
+    DEFAULT_OS=set([subdir.split("-")[0] for subdir in DEFAULT_SUBDIRS])
+    for subdir in os.listdir(conda_build.config.croot):
+        if not subdir.split("-")[0] in DEFAULT_OS:
+            continue
```
Comment (beckermr) [inline recipe/conda_forge_ci_setup/upload_or_check_non_existence.py:176]: What is this if statement catching?

Comment (isuruf) [reply recipe/conda_forge_ci_setup/upload_or_check_non_existence.py:176]: There are lots of folders in `<conda-root>/conda-bld`. This removes anything that doesn't look like a subdir

**Comment by isuruf** [comment]
@beckermr, I changed the logic to check for target_platform.

File: recipe/conda_forge_ci_setup/upload_or_check_non_existence.py [hunk 2]
```diff
@@ -158,7 +158,7 @@ def upload_or_check(
     allowed_dist_names = get_built_distribution_names(recipe_dir, variant)
```
Comment (beckermr) [inline recipe/conda_forge_ci_setup/upload_or_check_non_existence.py:158]: I think this pulls all of the names that are built. So we can look through every subdir and make sure it is in this list.

Comment (beckermr) [reply recipe/conda_forge_ci_setup/upload_or_check_non_existence.py:158]: This check is already there: https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/118/files#diff-2c55b5e34665621d4e425217f0f28912R192

So i'd just grab everything and prune it later.

Comment (isuruf) [reply recipe/conda_forge_ci_setup/upload_or_check_non_existence.py:158]: It gives use the names, but not the subdirs that they are located in

Comment (beckermr) [reply recipe/conda_forge_ci_setup/upload_or_check_non_existence.py:158]: sure but that's fine - any package with the right name can be uploaded regardless of subdir

Comment (beckermr) [reply recipe/conda_forge_ci_setup/upload_or_check_non_existence.py:158]: also the recipes are fully rendered in that function, so the correct subdir could be returned too.

**Comment by beckermr** [comment]
Do we need to double check that target_subdir returns the full subdir path?

**Comment by isuruf** [comment]
It shouldn't right?

File: recipe/conda_forge_ci_setup/upload_or_check_non_existence.py [hunk 3]
```diff
@@ -42,7 +42,8 @@ def get_built_distribution_names(recipe_dir, variant):
     for m in skipped_distributions:
         print("{} configuration was skipped in build/skip.".format(m.name()))
 
-    return set([m.name() for m, _, _ in metas if not m.skip()])
+    subdirs = set([m.config.target_subdir for m, _, _ in metas if not m.skip()])
+    return set([m.name() for m, _, _ in metas if not m.skip()]), subdirs
```
Comment (beckermr) [inline recipe/conda_forge_ci_setup/upload_or_check_non_existence.py:46]: Sets may not be ordered. I think you need lists here.

Comment (beckermr) [reply recipe/conda_forge_ci_setup/upload_or_check_non_existence.py:46]: Instead of all combinations of names, we can just use the actual ones built.

Comment (isuruf) [reply recipe/conda_forge_ci_setup/upload_or_check_non_existence.py:46]: Why is an ordered list needed?

Comment (isuruf) [reply recipe/conda_forge_ci_setup/upload_or_check_non_existence.py:46]: Ah, I see what you are saying. I think that's a clean up for another PR.

**Comment by isuruf** [comment]
Thanks for the review

**Pull request merged**